include/nuttx/mmcsd.h

```diff
@@ -27,6 +27,47 @@
 
 #include <nuttx/config.h>
 
+#include <stdint.h>
+#include <nuttx/fs/ioctl.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/* mmcsd ioctl */
+
+#define MMC_RPMB_TRANSFER       _MMCSDIOC(0x0000)
+
+/* rpmb request */
+
+#define MMC_RPMB_WRITE_KEY      0x01
+#define MMC_RPMB_READ_CNT       0x02
+#define MMC_RPMB_WRITE          0x03
+#define MMC_RPMB_READ           0x04
+
+/****************************************************************************
+ * Public Types
+ ****************************************************************************/
+
+struct mmc_rpmb_frame_s
+{
+  uint8_t  stuff[196];
+  uint8_t  key_mac[32];
+  uint8_t  data[256];
+  uint8_t  nonce[16];
+  uint32_t write_counter;
+  uint16_t addr;
+  uint16_t block_count;
+  uint16_t result;
+  uint16_t req_resp;
+};
+
+struct mmc_rpmb_transfer_s
+{
+  unsigned int num_of_frames;
```

Comment:
```suggestion
  uint32_t num_of_frames;
```
Wouldn't it be more appropriate here to use a fixed-width type?

Reply:
the first struct mmc_rpmb_frame_s is defined by emmc spec, and shared by host driver and emmc firmware, so the fixed- width is required here. On the other hand, mmc_rpmb_transfer_s is shared by nuttx driver and user program, so it's free to use the unfixed-width type because both side always compile with the same compiler and option.
BTW, the same struct is defined by Linux too.

Reply:
Thanks for the clarification!